web/packages/teleport/src/TopBar/TopBarSideNav.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Link } from 'react-router-dom';
+import { Flex, Image, TopNav } from 'design';
+import { matchPath, useHistory } from 'react-router';
+import { Theme } from 'design/theme/themes/types';
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+import useTeleport from 'teleport/useTeleport';
+import { UserMenuNav } from 'teleport/components/UserMenuNav';
+import { useFeatures } from 'teleport/FeaturesContext';
+import cfg from 'teleport/config';
+import { useLayout } from 'teleport/Main/LayoutContext';
+import { logos } from 'teleport/components/LogoHero/LogoHero';
+
+import { Notifications } from 'teleport/Notifications';
+
+export function TopBar({ CustomLogo }: TopBarProps) {
+  const ctx = useTeleport();
+  const history = useHistory();
+  const features = useFeatures();
+  const { currentWidth } = useLayout();
+  const theme: Theme = useTheme();
+
+  // find active feature
+  const feature = features
+    .filter(feature => Boolean(feature.route))
+    .find(f =>
+      matchPath(history.location.pathname, {
```

Comment:
Nit:

```suggestion
  const feature = features
    .find(f =>
      f.route && matchPath(history.location.pathname, {
```

I know it doesn't matter _that_ much, but the thought of this being called over and over again and us repeating that extra unnecessary loop through the array on each render makes me uneasy. 😅